packages/contracts/contracts/plugin/PluginSetupProcessor.sol

```diff
@@ -516,30 +511,24 @@ contract PluginSetupProcessor is DaoAuthorizable {
 
         PluginState storage pluginState = states[pluginInstallationId];
 
-        bytes32 setupId = _getSetupId(
+        bytes32 preparedSetupId = _getPreparedSetupId(
             _params.pluginSetupRef,
             hashPermissions(_params.permissions),
             _params.helpersHash,
             _params.initData,
             PreparationType.Update
         );
 
-        validateSetupId(pluginInstallationId, setupId);
+        validateSetupId(pluginInstallationId, preparedSetupId);
```

Comment:
`validatePreparedSetupId`?

Reply:
Let's leave it as it is. In the end, it vaidates the setupId which can be useful in future to validate appliedId if needed.

Reply:
No, we should be explicit. It is a different ID that is validated. 
In `PluginState`, we store the `preparedSetupId`. We should also name `setupIdToBlockNumber` to `preparedSetupIdToBlockNumber`

Reply:
done